### PR TITLE
Add helpers for service node port range configuration

### PR DESCRIPTION
**Pull request by abhat**
This PR adds a helper API to library-go for observing changes to the service node port range configuration.

### CODE REVIEW

File: pkg/operator/configobserver/network/observe_network.go [hunk 1]
```diff
@@ -127,3 +127,27 @@ func validateCIDRs(in []string) error {
...
+	if err != nil {
+		recorder.Warningf("GetServiceNodePortRangeFailed", "error getting networks.%s/cluster: %v", configv1.GroupName, err)
+		return "", err
+	}
+
+	if network.Spec.ServiceNodePortRange == "" {
+		return "", nil
+	}
+
+	if _, err = utilnet.ParsePortRange(
```
Comment (squeed) [inline pkg/operator/configobserver/network/observe_network.go:147]: nit: this linebreak seems odd?

**Comment by squeed** [comment]
one very minor thing, otherwise looks good.

File: pkg/operator/configobserver/network/observe_network.go (hunk 1 above)
Comment (abhat) [reply pkg/operator/configobserver/network/observe_network.go]: fixed.

File: pkg/operator/configobserver/network/observe_network.go [hunk 2]
```diff
@@ -127,3 +127,26 @@ func validateCIDRs(in []string) error {
 	}
 	return nil
 }
+
+// GetServiceNodePortRange retrieves the ServiceNodePortRange for the cluster.
+func GetServiceNodePortRange(lister configlistersv1.NetworkLister, recorder events.Recorder) (string, error) {
```
Comment (sttts) [inline pkg/operator/configobserver/network/observe_network.go:132]: why public?

Comment (abhat) [reply pkg/operator/configobserver/network/observe_network.go:132]: this will get called from the cluster-kube-apiserver-operator's `ConfigObserver`.

**Pull request merged**